tests/cli/Cli/Console/HandleCest.php

```diff
@@ -29,7 +34,278 @@ class HandleCest
+        $I->assertEquals($expected, $actual);
+        $expected = 'Hello !';
+        $actual = $dispatcher->getReturnedValue();
+        $I->assertEquals($expected, $actual);
+
+        $console->handle(
+            [
+                'task' => 'main',
+                'action' => 'hello',
+                'World',
+                '######',
+            ]
+        );
+        $expected = 'main';
+        $actual = $dispatcher->getTaskName();
+        $I->assertEquals($expected, $actual);
+        $expected = 'hello';
+        $actual = $dispatcher->getActionName();
+        $I->assertEquals($expected, $actual);
+        $expected = ['World', '######'];
+        $actual = $dispatcher->getParams();
+        $I->assertEquals($expected, $actual);
+        $expected = 'Hello World######';
+        $actual = $dispatcher->getReturnedValue();
+        $I->assertEquals($expected, $actual);
+    }
+
+    /**
+     * Tests Phalcon\Cli\Console :: handle()
+     *
+     * @param CliTester $I
+     *
+     * @author Nathan Edwards <https://github.com/npfedwards>
+     * @since 2018-12-26
+     */
+    public function cliConsoleHandleModule(CliTester $I)
+    {
+        require_once dataFolder('fixtures/modules/backend/tasks/MainTask.php');
+        $I->wantToTest("Cli\Console - handle() - Modules");
+        $console = $this->newCliConsole();
+        $this->setNewCliFactoryDefault();
+        $console->setDI($this->container);
+        $console->registerModules(
+            [
+                "frontend" => [
+                    "className" => "Phalcon\\Test\\Modules\\Frontend\\Module",
+                    "path" => __DIR__ . "/../../../_data/fixtures/modules/frontend/Module.php",
+                ],
+                "backend" => [
+                    "className" => "Phalcon\\Test\\Modules\\Backend\\Module",
+                    "path" => __DIR__ . "/../../../_data/fixtures/modules/backend/Module.php",
+                ]
+            ]
+        );
+        $console->dispatcher->setNamespaceName("Phalcon\\Test\\Modules\\Backend\\Tasks");
+
+        $I->expectThrowable(new \Exception("Task Run"), function () use ($console) {
+            $console->handle([
+                "module" => "backend",
+                "action" => "throw"
+            ]);
+        });
+        $dispatcher = $console->dispatcher;
+        $expected = 'main';
+        $actual = $dispatcher->getTaskName();
+        $I->assertEquals($expected, $actual);
+        $expected = 'throw';
+        $actual = $dispatcher->getActionName();
+        $I->assertEquals($expected, $actual);
+        $expected = 'backend';
+        $actual = $dispatcher->getModuleName();
+        $I->assertEquals($expected, $actual);
+    }
+
+    /**
+     * Tests Phalcon\Cli\Console :: handle()
+     *
+     * @param CliTester $I
+     *
+     * @author Nathan Edwards <https://github.com/npfedwards>
+     * @since 2018-12-26
+     */
+    public function cliConsoleHandleEvents(CliTester $I)
```

Comment:
I would like to split this test into smaller ones if possible. It makes it easier to figure out what is tested and where. 

Perhaps create something in the lines of 

`cliConsoleHandleEventBoot`
`cliConsoleHandleEventBeforeStartModule`
etc.